homeworks/rostyslav.havuka_RostyslavHavuka/6-typeScript/src/index.ts

```diff
@@ -0,0 +1,247 @@
+  private width: number;
+
+  constructor(
+    private scoreElement: HTMLHeadingElement,
+    private cubeScoreElement: HTMLDivElement,
+    private keyElement: HTMLInputElement,
+    private cubeElement?: HTMLDivElement,
+    private progressBarElement?: HTMLDivElement,
+  ) { }
+
+  start() {
+    window.addEventListener('load', this.startKeysInterval())
+
+    this.subscribeOnKeyPress()
+
+    keyStart.addEventListener('click', (): any => {
+      if (this.gameState === false) {
+        this.width = 1;
+        this.progressBar();
+        this.progressBarElement.classList.remove('hidden')
+        clearInterval(this.intervalId)
+        this.intervalId = setInterval(() => {
+          this.subScoreTimer(this.score)
+          this.sizeDown()
+          this.width = 1;
+          this.progressBar();
+        }, 3000);
+      }
+      this.gameState = true
+    })
+
+    keyStop.addEventListener('click', (): any => {
+      this.gameState = false;
+      this.progressBarElement.classList.add('hidden')
+      clearInterval(this.intervalStatusBarId)
+      this.timerProgressBar = 0;
+      this.width = 99;
+      clearInterval(this.intervalId)
+      alert('your Score:' + this.score)
+      this.score = 100;
+      cubeScoreElement.innerText = this.score.toString()
+      this.scoreElement.innerText = '';
+    })
+
+    keyAgain.addEventListener('click', (): any => {
+      if (this.gameState === true) {
+        this.score = 100;
+        this.width = 1;
+        this.progressBar();
+        cubeScoreElement.innerText = this.score.toString()
+        this.scoreElement.innerText = '';
+        clearInterval(this.intervalId)
+        this.intervalId = setInterval(() => {
+          this.subScoreTimer(this.score)
+          this.sizeDown()
+          this.width = 1;
+          this.progressBar();
+        }, 3000);
+        this.gameState = true
+      }
+    })
+  }
+
+  private startKeysInterval(): any {
+    let possible: string = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz1234567890";
+    for (var i = 0; i < 1; i++)
+      this.currentKey = possible.charAt(Math.floor(Math.random() * possible.length));
+    keyElement.value = this.currentKey.toUpperCase()
+  }
+
+  private trueGame(): void {
+    if (this.gameState === true) {
+      this.startKeysInterval()
+      this.addScore()
+      this.setNewInterval()
+      this.sizeUp()
+      clearInterval(this.intervalStatusBarId)
+      this.width = 1;
+      this.timerProgressBar = 0;
+      this.progressBar()
+    } else {
+      alert('Please press "Start"')
+    }
+  }
+
+  private falseGame(): void {
+    if (this.gameState === true) {
+      this.subScore()
+      this.startKeysInterval()
+      this.setNewInterval()
+      this.sizeDown()
+      clearInterval(this.intervalStatusBarId)
+      this.width = 1;
+      this.timerProgressBar = 0;
+      this.progressBar()
+    } else {
+      alert('Please press "Start"')
+    }
+  }
+
+  private setKey(e: any): void {
```

Comment:
No `any`. I think this type is `InputEvent`. But still, the best interface will be `setKey(key: string)`. So please, update your method interface and logic inside to match the new changes